lambda_utils/src/wellcome_lambda_utils/s3_utils.py

```diff
@@ -53,3 +55,23 @@ def should_copy():
             Bucket=dst_bucket,
             Key=dst_key
         )
+
+
+def _extract_s3_event(record):
+    event_datetime = dateutil.parser.parse(record["eventTime"])
+
+    return {
+        "event_name": record["eventName"],
+        "event_time": event_datetime,
+        "bucket_name": record["s3"]["bucket"]["name"],
+        "object_key": record["s3"]["object"]["key"],
+        "size": record["s3"]["object"]["size"],
+        "versionId": record["s3"]["object"].get("versionId")
+    }
```

Comment:
There’s potentially a neater way to do this with something called _descriptors_ – I’ll show you both later.